src/pymmcore_widgets/_snap_button_widget.py

```diff
@@ -65,11 +66,40 @@ def __init__(
         if len(self._mmc.getLoadedDevices()) > 1:
             self.setEnabled(True)
 
+    @property
+    def button_text(self) -> str:
+        """Set the text of the snap button."""
+        return self._button_text
+
+    @button_text.setter
+    def button_text(self, text: str) -> None:
+        self.setText(text)
+        self._button_text = text
+
+    @property
+    def icon_size(self) -> int:
+        """Set the snap button icon size."""
+        return self._icon_size
+
+    @icon_size.setter
+    def icon_size(self, size: int) -> None:
+        self.setIconSize(QSize(size, size))
+        self._icon_size = size
+
+    @property
+    def icon_color(self) -> COLOR_TYPES:
+        """Set the snap button icon color."""
+        return self._icon_color
+
+    @icon_color.setter
+    def icon_color(self, color: COLOR_TYPES) -> None:
+        self.setIcon(icon(MDI6.camera_outline, color=color))
+        self._icon_color = color
```

Comment:
So these are all cases where I think the new public methods are unnecessary.  This widget already _is_ a QPushButton, and all you're doing with `button_text`, `icon_size`, etc..., is just turning around and calling Qt's `self.setSameDarnThing`.  In the other example, you had a compound `QWidget` that didn't have things like `setText`, etc... but here you do.  So, this is an example of where I wouldn't create any additional public API where one already exists.